store limit test has data race issue

## Flaky Test

### Which jobs are failing

CGO_ENABLED=1 go test -timeout=15m -tags deadlock -race -covermode=atomic -coverprofile=covprofile -coverpkg=./... github.com/tikv/pd/pkg/core/storelimit github.com/tikv/pd/pkg/mcs/tso/server github.com/tikv/pd/pkg/schedule/operator github.com/tikv/pd/pkg/storage/kv github.com/tikv/pd/pkg/utils/keyutil github.com/tikv/pd/server/api 

==================
WARNING: DATA RACE
Write at 0x00c00001adc8 by goroutine 16:
  github.com/tikv/pd/pkg/core/storelimit.TestFeedback.func1()
      /home/runner/work/pd/pd/pkg/core/storelimit/limit_test.go:137 +0xa4

Previous read at 0x00c00001adc8 by goroutine 15:
  github.com/tikv/pd/pkg/core/storelimit.TestFeedback()
      /home/runner/work/pd/pd/pkg/core/storelimit/limit_test.go:188 +0xee4
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.1/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.1/x64/src/testing/testing.go:1629 +0x47

Goroutine 16 (running) created at:
  github.com/tikv/pd/pkg/core/storelimit.TestFeedback()
      /home/runner/work/pd/pd/pkg/core/storelimit/limit_test.go:134 +0x5[25](https://github.com/tikv/pd/actions/runs/4779473234/jobs/8496585204?pr=6358#step:5:26)
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.1/x64/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.20.1/x64/src/testing/testing.go:16[29](https://github.com/tikv/pd/actions/runs/4779473234/jobs/8496585204?pr=6358#step:5:30) +0x47

Goroutine 15 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.20.1/x64/src/testing/testing.go:1629 +0x805
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.20.1/x64/src/testing/testing.go:2036 +0x8d
  testing.tRunner()
      /opt/hostedtoolcache/go/1.20.1/x64/src/testing/testing.go:1576 +0x216
  testing.runTests()
      /opt/hostedtoolcache/go/1.20.1/x64/src/testing/testing.go:20[34](https://github.com/tikv/pd/actions/runs/4779473234/jobs/8496585204?pr=6358#step:5:35) +0x87c
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.20.1/x64/src/testing/testing.go:1906 +0xb[44](https://github.com/tikv/pd/actions/runs/4779473234/jobs/8496585204?pr=6358#step:5:45)
  main.main()
      _testmain.go:84 +0x2fc
==================
--- FAIL: TestFeedback (1.63s)
    testing.go:14[46](https://github.com/tikv/pd/actions/runs/4779473234/jobs/8496585204?pr=6358#step:5:47): race detected during execution of test

### CI link
https://github.com/tikv/pd/actions/runs/4779473234/jobs/8496585204?pr=6358

### Reason for failure (if possible)
a variable, used for test only, is accessed by multiple goroutines and isn't protected.

### Anything else

--- a/pkg/core/storelimit/limit_test.go
+++ b/pkg/core/storelimit/limit_test.go
@@ -18,6 +18,7 @@ import (
 	"container/list"
 	"context"
 	"math/rand"
+	"sync/atomic"
 	"testing"
 	"time"
 
@@ -125,21 +126,22 @@ func TestFeedback(t *testing.T) {
 	}
 	// region size is 10GB, snapshot write limit is 100MB/s and the snapshot concurrency is 3.
 	// the best strategy is that the tikv executing queue equals the wait.
-	regionSize, limit, wait := int64(10000), int64(100), int64(4)
-	iter := 100
+	const regionSize, limit, wait = int64(10000), int64(100), int64(4)
+	var iter atomic.Int32 
+	iter.Store(100)
 	ops := make(chan int64, 10)
 	ctx, cancel := context.WithCancel(context.Background())
 
 	// generate the operator
 	go func() {
 		for {
-			if s.Available(regionSize, SendSnapshot, constant.Low) && iter > 0 {
-				iter--
+			if s.Available(regionSize, SendSnapshot, constant.Low) && iter.Load() > 0 {
+				iter.Add(-1)
 				size := regionSize - rand.Int63n(regionSize/10)
 				s.Take(size, SendSnapshot, constant.Low)
 				ops <- size
 			}
-			if iter == 0 {
+			if iter.Load() == 0 {
 				cancel()
 				return
 			}
@@ -185,7 +187,7 @@ func TestFeedback(t *testing.T) {
 			err := exec*wait - cost
 			queue.Remove(first)
 			s.Feedback(float64(err))
-			if iter < 5 {
+			if iter.Load() < 5 {
 				re.Greater(float64(s.GetCap()), float64(regionSize*(wait-2)))
 				re.Less(float64(s.GetCap()), float64(regionSize*wait))
 			}

--- a/pkg/core/storelimit/limit_test.go
+++ b/pkg/core/storelimit/limit_test.go
@@ -127,7 +127,7 @@ func TestFeedback(t *testing.T) {
 	// region size is 10GB, snapshot write limit is 100MB/s and the snapshot concurrency is 3.
 	// the best strategy is that the tikv executing queue equals the wait.
 	const regionSize, limit, wait = int64(10000), int64(100), int64(4)
-	var iter atomic.Int32 
+	var iter atomic.Int32
 	iter.Store(100)
 	ops := make(chan int64, 10)
 	ctx, cancel := context.WithCancel(context.Background())

